### ISSUE TITLE
Enhancement: Add ability to enter tags directly in each file/document

### ISSUE THREAD

**Issue by magicnemo** [pbek/QOwnNotes#379]
Enhancement: Add ability to enter tags directly in each file/document

If you look at the new app called "Bear" for iOS/Mac, you have ability to enter a tag in a document just by writing e.g. #Todo bla bla# (where last # is used for entering several words in the tag).

Useful both for storing info in each text file which tags it belongs to (in case of database failure) and also since its quicker.

**Comment by pbek** [comment]
Thank you for your suggestion, but this was discussed here several times. It would not go well together with the current tagging system, it would be hard to parse because `#tag name#` is a bit too simple and I quite don't like the idea to manipulate the users notes when a tag should for example be removed or moved. 

Why do you think that's a good idea and how should it work?

**Comment by pbek** [comment]
If you just want to tag a note by writing a text in your note your could achieve that with a script:
You could use this method <https://github.com/pbek/QOwnNotes/tree/develop/doc/scripting#tag-the-current-note> in a script like this:
<https://github.com/pbek/QOwnNotes/blob/develop/doc/scripting/execute-command-after-note-update.qml>

**Comment by pbek** [comment]
I will close this issue until there is more information.

**Comment by pbek** [comment]
## 17.09.6
- added the first alpha version of external note tagging support via the 
  scripting engine (for [Issue #323](https://github.com/pbek/QOwnNotes/issues/323), [Issue #338](https://github.com/pbek/QOwnNotes/issues/338), [Issue #379](https://github.com/pbek/QOwnNotes/issues/379), [Issue #472](https://github.com/pbek/QOwnNotes/issues/472), [Issue #530](https://github.com/pbek/QOwnNotes/issues/530) and [Issue #603](https://github.com/pbek/QOwnNotes/issues/603))
    - as soon as a script is activated that implements the new function
      `noteTaggingHook` note tagging will be handled by that function
    - please install the script [note-tagging.qml](https://github.com/pbek/QOwnNotes/blob/develop/doc/scripting/note-tagging.qml)
      to test the basic functionality and report in [Issue #530](https://github.com/pbek/QOwnNotes/issues/530).
        - your internal tag database and your notes might get modified 
        - don't use this script on your production note folder, please make a 
          copy of it and test it there!
    - following features should already work via the QOwnNotes user interface
        - initially importing tags like `@tag` from your notes and overwriting
          your current tag assignment
            - you will not loose your tags tree, just the former assignment to notes
            - you can still move tags into other tags
            - if more than one tag has the same name in your tag tree the 
              first hit will be assigned
        - adding a tag to a note will add the tag to the note text        
        - removing a tag from a note will remove the tag from the note text
        - removing of tags in the tag list will remove those tags from your notes
        - renaming of tags in the tag list will rename those tags in your notes
        - bulk tagging of notes in the note list will add those tags to your notes
        - bulk removing of tags from notes in the note list will remove those 
          tags from your notes
    - for more information please take a look at the [scripting documentation](http://docs.qownnotes.org/en/develop/scripting/README.html#noteTaggingHook)

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if it works for you with the script [note-tagging.qml](https://github.com/pbek/QOwnNotes/blob/develop/doc/scripting/note-tagging.qml)?
